### PR TITLE
Filter and Web.xml For Authentication

### PR DESCRIPTION
**Pull request by hannahsalameh**
This is just the filter and web.xml. When it gets merged, it will break the website because logging in does not work yet

### CODE REVIEW

**Comment by hannahsalameh** [comment]
can still be reviewed 

NOTE TO SELF: dont merge until login merged

File: src/main/webapp/web.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee
```
Comment (devonwasson) [inline src/main/webapp/web.xml:4]: I don't think a lot of the HTML files in this web.xml are named this way yet. We should't be adding this in this PR. Instead, we should just remove references to files that dont exist or are incorrectly named, and then add that code back in during the relevant PRs.

File: src/main/java/com/google/sps/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.*;
+import javax.servlet.*;
+import javax.servlet.http.*;
```
Comment (devonwasson) [inline src/main/java/com/google/sps/AuthenticationFilter.java:16]: It's typically not great to import *, just import whatever you need.

File: src/main/java/com/google/sps/AuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.sps.utils.UserUtils;
+import com.google.sps.utils.CookieUtils;
```
Comment (devonwasson) [inline src/main/java/com/google/sps/AuthenticationFilter.java:21]: cookieutils is not used and can be safely removed.

File: src/main/java/com/google/sps/AuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
```
Comment (devonwasson) [inline src/main/java/com/google/sps/AuthenticationFilter.java:13]: javax.servlet.http.HttpSession is not used, i believe

File: src/main/java/com/google/sps/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    */
+ 
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse res = (HttpServletResponse) response;
+ 
+        Cookie cookies[] = req.getCookies();
+        Entity userEntity = UserUtils.getUserFromCookie(cookies, datastore);
+ 
+        //check to make sure there is an active session and that its attached to a person
```
Comment (devonwasson) [inline src/main/java/com/google/sps/AuthenticationFilter.java:47]: "its" should be "it's"

File: src/main/webapp/web.xml (hunk 1 above)
Comment (hannahsalameh) [reply src/main/webapp/web.xml:4]: discussed offline!

**Pull request merged**